admin/automation/process.py

```diff
@@ -0,0 +1,141 @@
+            """
+        )
+        self.parser.add_argument(
+            "input",
+            type=dir_path,
+            help="Input directory, containing grant files",
+        )
+        self.parser.add_argument(
+            "-d",
+            "--debug",
+            action="store_true",
+            help="Print debugging information",
+        )
+        self.args = self.parser.parse_args()
+
+
+def dir_path(string):
+    if os.path.isdir(string):
+        return string
+    else:
+        raise NotADirectoryError(string)
+
+
+class Status(str, Enum):
+    SELECTED = "Selected"
+    RUNNING = "Running"
+    COMPLETE = "Complete"
+
+
+class Contact(BaseModel):
+    name: str
+    email: str
+    organisationName: str
+
+
+class Websites(BaseModel):
+    website: List[str]
+
+
+class Webpage(BaseModel):
+    sitename: Optional[str] = Field(default=None)
+    summary: str
+
+
+class PropertiesField(BaseModel):
+    webpage: Webpage
+
+
+class ProposalField(BaseModel):
+    websites: Websites
+    contact: Contact
+
+
+class Proposal(BaseModel):
+    properties: PropertiesField
+    proposal: ProposalField
+    status: Status
+
+
+class Fund(BaseModel):
+    proposals: List[Proposal]
+
+
+class Result(BaseModel):
+    name: Optional[str] = Field(default=None)
+    websites: List[str]
+    summary: str
+    status: Status
+    contact: Contact
+
+
+def main(input_dir: str):
+    if not os.path.isdir(input_dir):
+        print(f"Error: '{input_dir}' does not exist or is not a directory.")
+        exit(1)
+
+    proposals = []
+
+    for input_file in os.listdir(input_dir):
+        if not input_file.endswith(".json"):
+            continue
+
+        input_path = os.path.join(input_dir, input_file)
+
+        with open(input_path, "r") as f:
+            data = json.load(f)
+
+        try:
+            fund = Fund(**data["fund"])
+
+            # TODO: improve
+            for proposal in fund.proposals:
+                if proposal.properties.webpage.sitename is None:
+                    logger.debug(proposal)
+                    continue
+
+                if proposal.status != Status.RUNNING:
+                    continue
+
+                proposals.append(
+                    Result(
+                        name=proposal.properties.webpage.sitename,
+                        websites=proposal.proposal.websites.website,
+                        summary=proposal.properties.webpage.summary,
+                        contact=proposal.proposal.contact,
+                        status=proposal.status,
+                    )
+                )
+        except ValidationError as e:
+            logger.error(e)
+
+    content = [proposal.model_dump() for proposal in proposals]
+    json.dump(content, sys.stdout, indent=2)
+
+
+if __name__ == "__main__":
+    args = Cli().args
+
+    logger = logging.getLogger(__name__)
+    logging_level = logging.DEBUG if args.debug else logging.WARNING
+    logging.basicConfig(level=logging_level)
```

Comment:
Please do all that in the main function. This block should never have business logic, it's not idiomatic to expect any work being done here.